src/pages/api/restaurant-finder.ts

```diff
@@ -24,23 +27,25 @@ const handler = async (req: RestaurantFinderRequest, res: NextApiResponse) => {
         return;
     }
 
-    const result = await axios("https://overpass-api.de/api/interpreter", {
-        method: "POST",
-        // The body contains the query
-        // to understand the query language see "The Programmatic Query Language" on
-        // https://wiki.openstreetmap.org/wiki/Overpass_API#The_Programmatic_Query_Language_(OverpassQL)
-        data:
-            "data=" +
-            encodeURIComponent(
-                `[out:json][timeout:10];(node["amenity"="restaurant"](around:${data.radius * 1000},${data.lat},${data.lng}););out tags qt center;`,
-            ),
+    const result = await axios<RestaurantFinderResponse>("https://api.geoapify.com/v2/places", {
+        method: "GET",
+        params: {
+            categories: "catering.restaurant",
+            filter: `circle:${data.lng},${data.lat},${data.radius * 1000}`,
+            limit: 100,
```

Comment:
if u are shuffling, why not just limit to 15

Reply:
If the user not gonna change their lat lng or radius, it's gonna be the same 15 restaurants

Reply:
fair enough